src/Promise/ErrorHandler.php

```diff
@@ -79,19 +79,61 @@ public static function notify($error)
         try {
             \call_user_func(self::$callback, $error);
         } catch (\Exception $e) {
-            // We're already a last chance handler, throwing doesn't make sense, so use a real fatal
-            trigger_error(sprintf(
-                "An exception has been thrown from the promise error handler registered to %s::set().\n\n%s",
-                __CLASS__,
-                (string) $e
-            ), E_USER_ERROR);
+            self::triggerErrorHandler(
+                "An exception has been thrown from the promise error handler registered to"
+                . " AsyncInterop\\Promise\\ErrorHandler::set().",
+                $e
+            );
+        } catch (\Throwable $e) {
+            self::triggerErrorHandler(
+                "An exception has been thrown from the promise error handler registered to"
+                . " AsyncInterop\\Promise\\ErrorHandler::set().",
+                $e
+            );
+        }
+    }
+
+    private static function triggerErrorHandler($message, $error) {
+        // We're already a last chance handler, throwing doesn't make sense, so use E_USER_ERROR.
+        // E_USER_ERROR is recoverable by a handler set via set_error_handler, which might throw, too.
+
+        try {
+            \trigger_error(
+                $message . "\n\n" . (string) $error,
+                E_USER_ERROR
+            );
+        } catch (\Exception $e) {
+            self::panic($e);
+        } catch (\Throwable $e) {
+            self::panic($e);
+        }
+    }
+
+    private static function panic($error) {
+        // The set error handler did throw or not exist, PHP's error handler threw, no chance to handle the error
+        // gracefully at this time. PANIC!
+
+        // Print error information to STDERR to the reason for the program abortion can be found, but do not expose
```

Comment:
Sorry, shoud have been more specific. I know what you're trying to say, but that sentence didn't parse over here (could be my fault though) :)

> Print error information to STDERR to the reason for the program abortion can be found